### PR TITLE
Add some tests for HTTP handler

### PR DESCRIPTION
**Pull request by relistan**
Adds a bunch of tests for the http `handleImage()` handler. Also a little refactoring and cleanup.

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -92,6 +92,21 @@ func scaleForRequest(r *http.Request) (float64, error) {
 	return scale, nil
 }
 
+func pageForRequest(r *http.Request) (int64, error) {
+	// Let's first parse out some URL args and return errors if
+	// we got some bogus stuff.
+	if r.FormValue("page") == "" {
+		return -1, fmt.Errorf("Invalid page!")
+	}
+	page, err := strconv.ParseInt(r.FormValue("page"), 10, 32)
```
Comment (mihaitodor) [inline http.go]: `strconv.ParseInt("", 10, 32)` returns an error. Seems like the check above is redundant: https://play.golang.org/p/OPZWvlMuwM

File: http.go [hunk 2]
```diff
@@ -92,6 +92,21 @@ func scaleForRequest(r *http.Request) (float64, error) {
...
+	if err != nil || page < 1 {
```
Comment (mihaitodor) [inline http.go]: You can also use `ParseUint` to avoid this check: https://golang.org/pkg/strconv/#ParseUint

File: http_test.go [hunk 3]
```diff
@@ -1,11 +1,64 @@
...
+	}
+)
+
+// CopyFile copies the contents from src to dst using io.Copy.
+// If dst does not exist, CopyFile creates it with permissions perm;
+// otherwise CopyFile truncates it before writing.
+func CopyFile(dst, src string, perm os.FileMode) (err error) {
+	in, err := os.Open(src)
+	if err != nil {
+		return
```
Comment (mihaitodor) [inline http_test.go]: Does this compile? (same below 2x)

File: http_test.go [hunk 4]
```diff
@@ -26,3 +79,139 @@ func Test_urlToFilename(t *testing.T) {
...
+
+				body, err := ioutil.ReadAll(recorder.Result().Body)
+				So(err, ShouldBeNil)
+				So(len(body), ShouldBeGreaterThan, 1024)
+				So(recorder.Result().StatusCode, ShouldEqual, 200)
+				So(didDownload, ShouldBeTrue)
+				So(downloadCount, ShouldEqual, 1)
+			})
+
+			Convey("Handles a jpegs", func() {
```
Comment (mihaitodor) [inline http_test.go]: Typo "a jpegs"

File: http_test.go [hunk 5]
```diff
@@ -26,3 +79,139 @@ func Test_urlToFilename(t *testing.T) {
...
+				h.handleImage(recorder, req)
+
+				body, err := ioutil.ReadAll(recorder.Result().Body)
+				So(err, ShouldBeNil)
+				So(len(body), ShouldBeGreaterThan, 1024)
+				So(recorder.Result().StatusCode, ShouldEqual, 200)
+				So(recorder.Result().Header["Content-Type"][0], ShouldEqual, "image/jpeg")
+			})
+
+			Convey("Handles a pngs", func() {
```
Comment (mihaitodor) [inline http_test.go]: Same

File: http_test.go [hunk 6]
```diff
@@ -26,3 +79,139 @@ func Test_urlToFilename(t *testing.T) {
...
+				req := httptest.NewRequest("GET", "/documents/somewhere/sample.pdf?page=1&width=1024&quality=75&imageType=image/png", nil)
```
Comment (mihaitodor) [inline http_test.go]: Somehow, I am surprised that the slash in `image/png` doesn't need to be URL encoded.

**Comment by relistan** [comment]
This compiles for me... and compiles on travis

**Comment by mihaitodor** [comment]
Ah, Oops, that named return parameter feature always gets me :)) Thanks :)

File: http_test.go (hunk 3 above)
Comment (mihaitodor) [reply http_test.go]: Oops, nevermind, it works as it is :)

**Pull request merged**